homeassistant/components/suez_water/sensor.py

```diff
@@ -57,11 +56,20 @@ async def async_setup_entry(
 ) -> None:
     """Set up Suez Water sensor from a config entry."""
     coordinator = entry.runtime_data
-    counter_id = entry.data[CONF_COUNTER_ID]
 
-    async_add_entities(
-        SuezWaterSensor(coordinator, counter_id, description) for description in SENSORS
-    )
+    def _check_device(known_device_id: None | str = None) -> None:
+        current_device_id = coordinator.data.current_device_id
+        if coordinator.data.new_device and (
+            known_device_id is None or current_device_id != known_device_id
+        ):
+            known_device_id = current_device_id
+            async_add_entities(
+                SuezWaterSensor(coordinator, coordinator.data.new_device, description)
+                for description in SENSORS
+            )
```

Comment:
so when the known device id changes we create entities, but when does the known device id change?